extensions/map-layers-formats/src/internal/GoogleMapsUtils.ts

```diff
@@ -0,0 +1,114 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { IModelApp } from "@itwin/core-frontend";
+import { GoogleMapsMapLayerFormat } from "../GoogleMaps/GoogleMapsImageryFormat";
+import { Logger } from "@itwin/core-bentley";
+import { ImageMapLayerProps, MapLayerProviderProperties } from "@itwin/core-common";
+import { Angle } from "@itwin/core-geometry";
+import { GoogleMapsCreateSessionOptions, GoogleMapsSession, ViewportInfo, ViewportInfoRequestParams } from "../GoogleMaps/GoogleMaps";
+
+const loggerCategory = "MapLayersFormats.GoogleMaps";
+
+/** @internal */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const GoogleMapsUtils = {
+  /**
+   * Creates a Google Maps session.
+   * @param apiKey Google Cloud API key
+   * @param opts Options to create the session
+   * @internal
+  */
+  createSession: async (apiKey: string, opts: GoogleMapsCreateSessionOptions): Promise<GoogleMapsSession> => {
+    const url = `https://tile.googleapis.com/v1/createSession?key=${apiKey}`;
+    const request = new Request(url, {method: "POST", body: JSON.stringify(opts)});
+    const response = await fetch (request);
+    if (!response.ok) {
+      throw new Error(`CreateSession request failed: ${response.status} - ${response.statusText}`);
+    }
+    Logger.logInfo(loggerCategory, `Session created successfully`);
+    return response.json();
+  },
+
+  /**
+   * Register the google maps format if it is not already registered.
+   * @internal
+  */
+  registerFormatIfNeeded: () => {
+    if (!IModelApp.mapLayerFormatRegistry.isRegistered(GoogleMapsMapLayerFormat.formatId)) {
+      IModelApp.mapLayerFormatRegistry.register(GoogleMapsMapLayerFormat);
+    }
+  },
+
+  /**
+ * Creates a Google Maps layer props.
+ * @param name Name of the layer (Defaults to "GoogleMaps")
+ * @param opts Options to create the session  (Defaults to satellite map type, English language, US region, and roadmap layer type)
+ * @internal
+*/
+  createMapLayerProps: (name: string = "GoogleMaps", opts?: GoogleMapsCreateSessionOptions): ImageMapLayerProps => {
+    GoogleMapsUtils.registerFormatIfNeeded();
```

Comment:
should we just throw instead if the format isnt registered? users should be calling MapLayersFormats.initialize right?

Reply:
yea we can just throw.  I was unsure about this one, wanted to make it easier for newcomers but probably not necessary.